### ISSUE TITLE
Typo in __GNU*C*_MINOR__ guard affecting gcc 3.x

### ISSUE THREAD

**Issue by 674c16e4-b864-4d55-874b-0db106acf296** [python/cpython#74122]
BPO | [29936](https://bugs.python.org/issue29936)
--- | :---
Nosy | @tiran, @benjaminp, @niklasf
PRs | <li>python/cpython#878</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-03-29.04:58:52.802>
created_at = <Date 2017-03-28.16:26:04.441>
labels = ['build', '3.7']
title = 'Typo in __GNU*C*_MINOR__ guard affecting gcc 3.x'
updated_at = <Date 2017-03-29.04:58:52.801>
user = 'https://github.com/niklasf'
```

bugs.python.org fields:
```python
activity = <Date 2017-03-29.04:58:52.801>
actor = 'benjamin.peterson'
assignee = 'none'
closed = True
closed_date = <Date 2017-03-29.04:58:52.802>
closer = 'benjamin.peterson'
components = ['Build']
creation = <Date 2017-03-28.16:26:04.441>
creator = 'niklasf'
dependencies = []
files = []
hgrepos = []
issue_num = 29936
keywords = []
message_count = 3.0
messages = ['290738', '290767', '290768']
nosy_count = 3.0
nosy_names = ['christian.heimes', 'benjamin.peterson', 'niklasf']
pr_nums = ['878']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'compile error'
url = 'https://bugs.python.org/issue29936'
versions = ['Python 3.7']
```

</p></details>

**Comment by 674c16e4-b864-4d55-874b-0db106acf296** [comment]
The patch in http://bugs.python.org/issue16881 disables the nicer macro for gcc 3.x due to a small typo.

The build is not failing. The guard just unnescessarily evaluates to false.

**Comment by benjaminp** [comment]
New changeset 83371f4f7f70406ea5d08c5fa8dbdbbcc0355ee1 by Benjamin Peterson (Niklas Fiekas) in branch 'master':
bpo-29936: fix typo \_\_GNU\*C*_MINOR__ (#878)
https://github.com/python/cpython/commit/83371f4f7f70406ea5d08c5fa8dbdbbcc0355ee1

**Comment by benjaminp** [comment]
Thanks. We should probably just unsupport gcc 3.x.
